Behaviors/FrameworkElementBehaviors.cs

```diff
@@ -0,0 +1,96 @@
+        /// <param name="value">The <see cref="FrameworkElement"/> to be added to the <see cref="RegisteredChildrenProperty"/>.</param>
+        public static void SetRegisteredChildren(FrameworkElement element, FrameworkElement value)
+        {
+            var children = GetRegisteredChildren(element);
+            // need a copy of the list, otherwise the callback does not get called
+            var newChildren = new List<FrameworkElement>(children);
+            newChildren.Add(value);
+            element.SetValue(RegisteredChildrenProperty, newChildren);
+        }
+
+        /// <summary>
+        /// Gets the <see cref="RegisteredChildrenProperty"/> as a <see cref="List{T}"/> of <see cref="FrameworkElement"/>s.
+        /// </summary>
+        /// <param name="element">The <see cref="FrameworkElement"/> from which the <see cref="RegisteredChildrenProperty"/> is being retrieved.</param>
+        /// <returns>The <see cref="RegisteredChildrenProperty"/> as a <see cref="List{T}"/></returns>
+        public static List<FrameworkElement> GetRegisteredChildren(FrameworkElement element) 
+        {
+            var collection = (List<FrameworkElement>)element.GetValue(RegisteredChildrenProperty);
+            if (collection == null)
+            {
+                collection = new List<FrameworkElement>();
+                element.SetValue(RegisteredChildrenProperty, collection);
+            }
+            return collection; 
+        }
+
+        /// <summary>
+        /// A <see cref="PropertyChangedCallback"/> that handles <see cref="RegisteredChildrenProperty"/> changes.
+        /// It sets up a <see cref="FrameworkElement"/>s properties to be binded to its registered children properties. 
+        /// </summary>
+        /// <param name="obj">The <see cref="DependencyObject"/> on which the <see cref="RegisteredChildrenProperty"/> has changed.</param>
+        /// <param name="args">The <see cref="DependencyPropertyChangedEventArgs"/> for <see cref="RegisteredChildrenProperty"/> changed events.</param>
+        private static void OnRegisteredChildrenChanged(DependencyObject obj, DependencyPropertyChangedEventArgs args)
+        {
+            if (!(obj is FrameworkElement element))
+                return;
+
+            if (element.IsLoaded)
+            {
+                SetupVisibilityBinding(element);
+            }
+            else
+            {
+                element.Loaded += (sender, eventArgs) => SetupVisibilityBinding(element);
+            }
+        }
+
+        /// <summary>
+        /// Adds and optionally clears the <see cref="Visibility"/> property binding on a <see cref="FrameworkElement"/>.
+        /// </summary>
+        /// <param name="element">The <see cref="FrameworkElement"/> on which the binding will be added and optionally removed.</param>
+        private static void SetupVisibilityBinding(FrameworkElement element)
```

Comment:
Nitpicking on naming: the property names are RegisterParent and RegisterChild, but they impact visibility, and visibility alone, which would be hard to guess without reading the internal code of the properties. Either the property names should hint at visibility (<groubox IsVisibilityContainer=true> and <button VisibilityContainer={binding ...}>) or the visibility should be something that makes use of the relationship and is applied separately somehow.

Reply:
I'll look into it!

Reply:
My proposal:

Keep the RegisteredChildren and RegisteredParent property and for each behavior we implement we add a boolean property. In this situation we want to impact the `Visibility` property, so I would add a `HasVisibilityChildren` property and set it to true via XAML. Then in   the `OnRegisteredChildrenChanged` method we check if the flag is set. If it is, execute the Visibility binding method, if it's not, skip that specific behavior.

If we get to, let's say, 6-7 behaviors in the future, instead of if-elseing every property we can create a private Dictionary<Behavior, bool> used to "bind" behaviors and a boolean value indicating whether the behavior should be executed or not.

What do you think?

Reply:
Updated the naming of the properties.